### PR TITLE
Support relative paths in import and export

### PR DESCRIPTION
**Pull request by zhiltsov-max**
Depends on #1715

### Motivation and context

Added support for importing annotations with relative paths.
Fixed file name matching when importing annotations with relative paths and the target task has been created with images from file share.

How to test:
- create a task with relative paths
- import and export annotations. Images with same names lying in different directories should be handled normally.

### How has this been tested?

Unit tests

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by nmanovic** [comment]
@zhiltsov-max , really huge changes. Let's organise a review session (95 files!). Could you please schedule a meeting? If we spend 2 minute per file, we need 3 hours to discuss the PR.

**Comment by zhiltsov-max** [comment]
@nmanovic, can we stop counting changed files, please? Most of the files are added examples of the formats in tests, it hardly complicates the review.

**Comment by azhavoro** [comment]
@zhiltsov-max could you please fix conflicts?

File: datumaro/datumaro/plugins/yolo_format/extractor.py [hunk 1]
```diff
@@ -99,25 +100,38 @@ def __init__(self, config_path, image_info=None):
...
+
+    @classmethod
+    def name_from_path(cls, path):
+        path = cls.localize_path(path)
+        parts = split_path(path)
+        if 1 < len(parts) and not osp.isabs(path):
+            # NOTE: when path is like [data/]<subset_obj>/<image_name>
+            # drop everything but <image name>
+            # <image name> can be <a/b/c/filename.ext>, so no just basename()
+            path = osp.join(*parts[1:])
```
Comment (nmanovic) [inline datumaro/datumaro/plugins/yolo_format/extractor.py:123]: Codacy Issue found: [No value for argument 'a' in function call](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5416079)

**Comment by nmanovic** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- datumaro/tests/test_yolo_format.py  2
- cvat/apps/dataset_manager/bindings.py  1
- cvat/apps/dataset_manager/formats/yolo.py  2
         

Complexity decreasing per file
==============================
+ datumaro/datumaro/plugins/voc_format/converter.py  -2
+ datumaro/datumaro/plugins/tf_detection_api_format/extractor.py  -1
+ datumaro/datumaro/plugins/yolo_format/converter.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5416079&bid=17757343)

File: cvat/apps/dataset_manager/tests/_test_formats.py [hunk 2]
```diff
@@ -397,3 +401,96 @@ def load_dataset(src):
                     self.assertEqual(len(dataset), task["size"])
                 self._test_export(check, task, format_name, save_images=False)
 
+class FrameMatchingTest(_DbTestBase):
+    def _generate_task_images(self, paths):
```
Comment (nmanovic) [inline cvat/apps/dataset_manager/tests/_test_formats.py:405]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5416079)

File: datumaro/datumaro/plugins/yolo_format/extractor.py (hunk 1 above)
Comment (zhiltsov-max) [reply datumaro/datumaro/plugins/yolo_format/extractor.py:123]: Wrong

**Comment by zhiltsov-max** [comment]
@azhavoro, 

Problem 1 - should be resolved by #1615 and #1728
Problem 2 - resolved in #1638

**Comment by nmanovic** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- cvat/apps/dataset_manager/formats/yolo.py  2
- cvat/apps/dataset_manager/bindings.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5416079&bid=17757343)

File: cvat/apps/dataset_manager/tests/_test_formats.py (hunk 2 above)
Comment (nmanovic) [inline cvat/apps/dataset_manager/tests/_test_formats.py:405]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5416079)

**Comment by nmanovic** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- cvat/apps/dataset_manager/formats/yolo.py  2
- cvat/apps/dataset_manager/bindings.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5416079&bid=17757343)

**Comment by nmanovic** [comment]
@zhiltsov-max , could you please resolve conflict?
@azhavoro , could you please test the patch ASAP? Need to merge it till end of week.

**Comment by nmanovic** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- cvat/apps/dataset_manager/formats/yolo.py  2
- cvat/apps/dataset_manager/bindings.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5416079&bid=17757343)

**Comment by azhavoro** [comment]
> @azhavoro,
> 
> Problem 1 - should be resolved by #1615 and #1728
> Problem 2 - resolved in #1638



> @azhavoro,
> 
> Problem 1 - should be resolved by #1615 and #1728
> Problem 2 - resolved in #1638

@zhiltsov-max uploading of MOT still doesn't work for me.

**Comment by zhiltsov-max** [comment]
@azhavoro, could you share the MOT annotations?

**Comment by azhavoro** [comment]
> @azhavoro, could you share the MOT annotations?

[task_task-2020_06_19_09_15_02-mot 1.1.zip](https://github.com/opencv/cvat/files/4803604/task_task-2020_06_19_09_15_02-mot.1.1.zip)
[task_task-2020_06_19_09_15_02-cvat for images 1.1.zip](https://github.com/opencv/cvat/files/4803605/task_task-2020_06_19_09_15_02-cvat.for.images.1.1.zip)

**Pull request merged**